### PR TITLE
Java: Promote Hardcoded JWT credential query

### PR DESCRIPTION
**Pull request by egregius313**
Query promotion for the experimental `java/hardcoded-jwt-key` by integrating it into `java/hardcoded-credential-api-call`.

Promotion of #9036.

### CODE REVIEW

**Review by jcogs33 — COMMENTED** [review]
Hey Ed, I’ve added a few comments/questions below, but I’ll leave a full review and approval to someone else. 🙂 

- This will probably need a change note under `lib` for the new sinks. (FYI, it’s standard to mention the original author of the experimental query in the change note when doing a promotion, see [this discussion](https://github.com/github/codeql/pull/10785#discussion_r1012808835))
- Where is the `java/ql/src/Security/CWE/CWE-798/HardcodedJwtKey.java` file being used? Was this meant to be added as an example in `HardcodedCredentialsApiCall.qhelp`?
- Would it make sense to include test cases for all of the new models? e.g. HMAC384 and HMAC512 as well?

File: java/ql/test/experimental/query-tests/security/CWE-321/HardcodedJwtKey.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ public String accessTokenBad(String username) {
                 .withExpiresAt(new Date(new Date().getTime() + ACCESS_EXPIRE_TIME))
                 .withIssuer(ISSUER)
                 .withClaim("username", username)
-                .sign(algorithm);
+                .sign(algorithm); // $ HardcodedCredentialsApiCall
```
[the remark this answers is not in the document]
Comment (egregius313) [reply java/ql/test/experimental/query-tests/security/CWE-321/HardcodedJwtKey.java]: I had forgotten to get around to removing the files. The files under the `experimental` folder have been removed.

**Review by atorralba — COMMENTED** [review]
The `HardcodedJwtKey.java` file under `Security/CWE/CWE-798` isn't doing much unless you reference it in the QHelp. So I'd recommend doing that, or just removing it.

But otherwise this looks good to me! 👍

**Review by atorralba — APPROVED** [review]
DCA looks good! 👍

**Pull request merged**